src/multirust-utils/src/raw.rs

```diff
@@ -355,16 +355,50 @@ pub fn remove_dir(path: &Path) -> io::Result<()> {
         // The implementation of `remove_dir_all` is broken on windows,
         // so may need to try multiple times!
         for _ in 0..5 {
-            result = fs::remove_dir_all(path);
+            result = rm_rf(path);
             if !is_directory(path) {
                 return Ok(());
             }
-            thread::sleep(Duration::from_millis(100));
+            thread::sleep(Duration::from_millis(16));
         }
         result
     }
 }
 
+// Again because remove_dir all doesn't delete write-only files on windows,
+// this is a custom implementation, more-or-less copied from cargo.
+// cc rust-lang/rust#31944
+// cc https://github.com/rust-lang/cargo/blob/master/tests/support/paths.rs#L52-L80
+fn rm_rf(path: &Path) -> io::Result<()> {
+    if path.exists() {
+        for file in fs::read_dir(path).unwrap() {
+            let ref file = try!(file).path();
+
+            if file.is_dir() {
```

Comment:
I've actually seen quite noticeable speed improvements in the past from using `is_dir` from the `DirEntry` rather than running another stat call here, perhaps the `is_dir` could be loaded from `try!(file.file_type()).is_dir()`?